modules/apps/batch-planner/batch-planner-web/src/main/resources/META-INF/resources/js/import/ImportForm.js

```diff
@@ -128,12 +184,24 @@ function ImportForm({
 				setMappingsToBeEvaluated(template.mappings);
 			}
 		}
+		function handlesFileFormattedContent({fileContent}) {
+			setFileContent(fileContent);
+		}
+		function handleCsvHeaders({csvContainsHeaders}) {
+			setCsvHeaders(csvContainsHeaders);
+		}
+
+		function handleCsvHeaders({csvContainsHeaders}) {
+			setCsvHeaders(csvContainsHeaders);
+		}
 
+		Liferay.on(CSV_HEADERS, handleCsvHeaders);
 		Liferay.on(FILE_SCHEMA_EVENT, handleFileSchemaUpdate);
 		Liferay.on(SCHEMA_SELECTED_EVENT, handleSchemaUpdated);
 		Liferay.on(TEMPLATE_SELECTED_EVENT, handleTemplateSelect);
 
 		return () => {
+			Liferay.detach(CSV_HEADERS, handleCsvHeaders);
```

Comment:
As mentioned before, the ImportForm component should not be aware of any of the supported formats. It should receive data already parse in only one format.